### PR TITLE
[do not merge][autoscaler] Fix up UX

### PR DESCRIPTION
**Pull request by richardliaw**
## What do these changes do?

This is still work in progress.

TODOs before able to try out:

 - [ ] Setup autoscaler logfile
 - [ ] Check and test for log output of `ray --autoscaler-config`

## Linter

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.

### CODE REVIEW

File: python/ray/autoscaler/updater.py [hunk 1]
```diff
@@ -159,20 +160,23 @@ def run(self):
...
-                logger.debug("NodeUpdater: "
-                             "{}: Waiting for SSH...".format(self.node_id))
-
                 # Setting redirect=False allows the user to see errors like
                 # unix_listener: path "/tmp/rkn_ray_ssh_sockets/..." too long
                 # for Unix domain socket.
-                self.ssh_cmd("uptime", connect_timeout=5, redirect=False)
-
+                logger.info(waiting_msg + "." * retries)
+                with open("/dev/null") as devnull:
```
Comment (ericl) [inline python/ray/autoscaler/updater.py]: We cannot simply redirect things to /dev/null.

File: python/ray/autoscaler/updater.py [hunk 2]
```diff
@@ -204,7 +210,10 @@ def sync_file_mounts(self, sync_cmd):
                     "mkdir -p {}".format(os.path.dirname(remote_path)),
                     redirect=None,
                 )
-                sync_cmd(local_path, remote_path, redirect=None)
+                with open("/dev/null") as redirect:
```
Comment (ericl) [inline python/ray/autoscaler/updater.py]: Same comment as above.

**Comment by richardliaw** [comment]
Yeah the fix here is to make the log file global and point it there

File: python/ray/autoscaler/updater.py (hunk 1 above)
Comment (simon-mo) [reply python/ray/autoscaler/updater.py]: Recommending [redict_stdout](https://docs.python.org/3.4/library/contextlib.html#contextlib.redirect_stdout)

```
f = io.StringIO()
with redirect_stdout(f):
    help(pow)
s = f.getvalue()
```

**Pull request closed without merging**